pkgs/development/tools/analysis/ikos/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, lib, fetchFromGitHub, cmake, boost
+, gmp, llvmPackages_7, sqlite, python3
```

Comment:
You should just take an `llvmPackages` argument here, and then set that to `llvmPackages_7` in all-packages.nix, like [this](https://github.com/NixOS/nixpkgs/blob/ad3e9191d16722ea3eec32f4cd689eea730f39f6/pkgs/top-level/all-packages.nix#L8466-L8468).

Additionally, you're currently referencing two different `stdenv`s – the normal one, and `llvmPackages_7.stdenv`. You could consider passing the LLVM packages in as an override from all-packages.nix to make this a little less confusing.

All together, it'd look something like this:

```nix
callPackage ../development/tools/analysis/ikos {
  inherit (llvmPackages_7) stdenv clang llvm;
}
```

That way, your package definition isn't referencing a specific version of LLVM. We generally try to avoid that.